git_workflow.md

```diff
@@ -0,0 +1,222 @@
+
+6. Stage and Commit Code <a name="6"/>
+
+"A commit should be a wrapper for related changes." Read these [GitHub best commit practices](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices) that we follow.
+
+In general, you can stage multiple changes to save and track your work, but a commit should only be pushed to GitHub when you are done working on a logical chunk of code. This is so the code can easily be rolled back. This doesn't mean your changes are huge, but rather that you commit based on the completion of a fix or enhancement into logical chunks. For example, if your mission is to write unit tests for all functions in one .R file, you could have a commit after you complete tests for one or two functions.
+
+## Good Commit Messages
+
+Why good messages are important:
+[Erlang: Writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages): "Good commit messages serve at least three important purposes:
+- To speed up the reviewing process.
+- To help us write a good release note.
+- To help the future maintainers of Erlang/OTP (it could be you!), say five years into the future, to find out why a particular change was made to the code or why a specific feature was added."
+
+How to write good messages:
+[Who-T: On commit messages](http://who-t.blogspot.com/2009/12/on-commit-messages.html): "A good commit message should answer three questions about a patch:
+- Why is it necessary? It may fix a bug, it may add a feature, it may improve performance, reliabilty, stability, or just be a change for the sake of correctness.
+- How does it address the issue? For short obvious patches this part can be omitted, but it should be a high level description of what the approach was.
+- What effects does the patch have? (In addition to the obvious ones, this may include benchmarks, side effects, etc.)"
+
+Rules for commit messages:
+- Begin message with a short summary (< 50 character)
+- Press enter
+- Have bullet points that outline specifics including reasoning for specific changes
+
+An example of a good commit message [commit 0d6577](https://github.com/DrylandEcology/rSFSW2/pull/306/commits/14c6fe45729fb288cb5ec5438b51d80db60d6577):
+
+`Prepare, use, and populate the temporary output databases
+
+- function `run_simulation_experiment` is now calling
+`make_temporary_dbOutputs` before running the simulation runs
+
+- function `move_output_to_dbOutput` gained a new argument
+`use_dbTempOut` that is by default set to true`
+
+## Stage and Commit Workflow:
+
+* __Stage__ your changes in a snapshot: `git add <file>` or `git add <directory>` or `git add -p`
```

Comment:
Maybe add that several GUIs allow staging per line instead of per file

Reply:
which is helpful to split up a bunch of development work into logical commit units